api/api/views/evaluate_team_set.py

```diff
@@ -0,0 +1,94 @@
+from typing import Dict, Any
+
+from rest_framework import viewsets
+from rest_framework.decorators import action
+from schema import SchemaError
+
+from api.api.utils.response_with_metadata import ResponseWithMetadata
+from api.api.validators.evaluate_team_set_validator import EvaluateTeamSetValidator
+from api.models.team_set.serializer import TeamSetSerializer
+from benchmarking.evaluations.interfaces import TeamSetMetric
+from benchmarking.evaluations.metrics.average_project_requirements_coverage import (
+    AverageProjectRequirementsCoverage,
+)
+from benchmarking.evaluations.metrics.average_social_satisfied import (
+    AverageSocialSatisfaction,
+)
+from benchmarking.evaluations.metrics.average_solo_status import AverageSoloStatus
+from benchmarking.evaluations.metrics.average_timeslot_coverage import (
+    AverageTimeslotCoverage,
+)
+from benchmarking.evaluations.metrics.cosine_similarity import AverageCosineDifference
+from benchmarking.evaluations.metrics.utils.team_calculations import (
+    is_happy_team_allhp_friend,
+)
+
+
+class EvaluateTeamSetViewSet(viewsets.GenericViewSet):
+    data_label = "metric_evaluations"
+
+    @action(url_path="team_set", detail=False, methods=["POST"])
+    def evaluate_team_set(self, request):
+        try:
+            request_data: Dict[str, Dict[str, Any]] = dict(request.data)
+
+            EvaluateTeamSetValidator(request_data).validate()
+
+            team_set_json = request_data["team_set"]
+            team_set = TeamSetSerializer().decode(team_set_json)
+            metrics = request_data["metrics"]
+
+            return_metrics: Dict[str, float] = {
```

Comment:
it might be useful to return the team set as well as the metrics.

This way you don't need to match a specific request with a response to know what it was referring to. (Consider the async case where you send multiple teamsets for evaluation at the same time)

Just a thought though

Reply:
`ResponseWithMetadata` is a bit shaky of a paradigm for this reason, too. We should have the freedom to return multiple keys of data freely without the data label forcing us to return only one

Reply:
I can put a TODO